test/debug/config_test.rb

```diff
@@ -303,4 +303,31 @@ def test_p_with_keep_alloc_site
       end
     end
   end
+
+  class LogLevelTest < TestCase
+    def program
+      <<~RUBY
+      1| Thread.new { puts(1) }.join
+     2| binding.b
+      RUBY
+    end
+
+    def test_debugger_takes_log_levels_config
+      # default WARN level doesn't report threads creation
+      debug_code(program, remote: false) do
+        type 'c'
+        assert_no_line_text(/Thread #\d+ is created/)
+        type 'c'
+      end
+
+      ENV["RUBY_DEBUG_LOG_LEVEL"] = "INFO"
+      debug_code(program, remote: false) do
```

Comment:
Because the current logging logic only uses standard output/error, it's complicated to test in remote modes.